pkg/registry/helm.go

```diff
@@ -0,0 +1,363 @@
+	}
+
+	return h, nil
+}
+
+// RegistrySpecDir is the registry directory.
+func (h *Helm) RegistrySpecDir() string {
+	return h.Name()
+}
+
+// RegistrySpecFilePath is the path for the registry.yaml
+// NOTE: this function appears to be github registry specific and may not
+// need to be a part of the interface.
+func (h *Helm) RegistrySpecFilePath() string {
+	return ""
+}
+
+// FetchRegistrySpec fetches the registry spec. This method returns an unmarshalled version
+// of registry.yaml
+func (h *Helm) FetchRegistrySpec() (*Spec, error) {
+	spec := &Spec{
+		Libraries: LibraryRefSpecs{},
+	}
+
+	repository, err := h.repositoryClient.Repository()
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving repository")
+	}
+
+	for _, chart := range repository.Latest() {
+		name := chart.Name
+		if name == "" {
+			return nil, errors.Errorf("entries are invalid")
+		}
+
+		spec.Libraries[name] = &LibraryRef{
+			Path:    name,
+			Version: chart.Version,
+		}
+	}
+
+	return spec, nil
+}
+
+// MakeRegistryRefSpec returns app registry ref spec.
+func (h *Helm) MakeRegistryRefSpec() *app.RegistryRefSpec {
+	return h.spec
+}
+
+// ResolveLibrarySpec returns a resolved spec for a part.
+func (h *Helm) ResolveLibrarySpec(partName, version string) (*parts.Spec, error) {
+	chart, err := h.repositoryClient.Chart(partName, version)
+	if err != nil {
+		return nil, errors.Wrapf(err, "retrieving chart %s-%s", partName, version)
+	}
+
+	part := &parts.Spec{
+		APIVersion: parts.DefaultAPIVersion,
+		Kind:       parts.DefaultKind,
+
+		Name:        chart.Name,
+		Version:     chart.Version,
+		Description: chart.Description,
+	}
+
+	return part, nil
+}
+
+// ResolveLibrary fetches the part and creates a parts spec and library ref spec.
+func (h *Helm) ResolveLibrary(partName string, partAlias string, version string, onFile ResolveFile, onDir ResolveDirectory) (*parts.Spec, *app.LibraryRefSpec, error) {
+	part, err := h.ResolveLibrarySpec(partName, version)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	chart, err := h.repositoryClient.Chart(partName, version)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "retrieving chart %s-%s", partName, version)
+	}
+
+	for _, u := range chart.URLs {
+		resp, err := h.httpGet(u)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			return nil, nil, errors.Errorf("unexpected HTTP status code %q when retrieving helm chart", resp.Status)
+		}
+
+		defer func() {
+			cerr := resp.Body.Close()
+			if cerr != nil {
+				err = cerr
+			}
+		}()
+
+		handler := func(f *archive.File) error {
+			var b []byte
+			b, err = ioutil.ReadAll(f.Reader)
+			if err != nil {
+				return err
+			}
+
+			name := path.Join(chart.Name, "helm", chart.Version, f.Name)
+
+			return onFile(name, b)
+		}
+
+		if err = h.unarchiver.Unarchive(resp.Body, handler); err != nil {
+			return nil, nil, err
+		}
+	}
+
+	refSpec := &app.LibraryRefSpec{
+		Name:     partAlias,
+		Registry: h.Name(),
```

Comment:
Conspicuously missing is `GitVersion` - should we discuss how we might better abstract away the version details while allowing them to be persisted in `app.yaml`?

Reply:
There is not GitVersion if there is not GitRepo. This is an implementation detailed that leaked before registries were understood.